src/compiler/compile-props.js

```diff
@@ -122,6 +123,9 @@ export function compileProps (el, propOptions, vm) {
     } else if ((value = getAttr(el, attr)) !== null) {
       // has literal binding!
       prop.raw = value
+    } else if (propsData && (value = propsData[attr]) !== null) {
```

Comment:
`attr` is hyphenated, but the user may use camelCase prop names in `propsData`. We need to check  both `propsData[attr]` and `propsData[name]`.